examples/pytorch/mnist-ddp/mnist.ipynb

```diff
@@ -0,0 +1,627 @@
+      "Train Epoch: 1 [49000/60000 (82%)]\tLoss: 0.373420\n",
+      "Train Epoch: 1 [50000/60000 (83%)]\tLoss: 0.295203\n",
+      "Train Epoch: 1 [51000/60000 (85%)]\tLoss: 0.474386\n",
+      "Train Epoch: 1 [52000/60000 (87%)]\tLoss: 0.411164\n",
+      "Train Epoch: 1 [53000/60000 (88%)]\tLoss: 0.359452\n",
+      "Train Epoch: 1 [54000/60000 (90%)]\tLoss: 0.309372\n",
+      "Train Epoch: 1 [55000/60000 (92%)]\tLoss: 0.356350\n",
+      "Train Epoch: 1 [56000/60000 (93%)]\tLoss: 0.363343\n",
+      "Train Epoch: 1 [57000/60000 (95%)]\tLoss: 0.482411\n",
+      "Train Epoch: 1 [58000/60000 (97%)]\tLoss: 0.353806\n",
+      "Train Epoch: 1 [59000/60000 (98%)]\tLoss: 0.274754\n",
+      "Training is finished\n"
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "\n",
+    "# Set the Torch Distributed env variables so the training function can be run in the notebook\n",
+    "# See https://pytorch.org/docs/stable/elastic/run.html#environment-variables\n",
+    "os.environ[\"RANK\"] = \"0\"\n",
+    "os.environ[\"LOCAL_RANK\"] = \"0\"\n",
+    "os.environ[\"WORLD_SIZE\"] = \"1\"\n",
+    "os.environ[\"MASTER_ADDR\"] = \"localhost\"\n",
+    "os.environ[\"MASTER_PORT\"] = \"1234\"\n",
+    "\n",
+    "# Run the training function locally\n",
+    "train_fashion_mnist({\"epochs\": 1})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create the Kubeflow Training Client"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "from kubeflow.training import Trainer, TrainingClient\n",
+    "client = TrainingClient()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## List the training runtimes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Runtime(name='torch-distributed', phase='pre-training', accelerator='Unknown', accelerator_count='Unknown')\n"
+     ]
+    }
+   ],
+   "source": [
+    "for runtime in client.list_runtimes():\n",
+    "    print(runtime)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run the distributed training Job"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "job_name = client.train(\n",
+    "    # Use one the of the training runtimes installed on your Kubernetes cluster\n",
+    "    runtime_ref=\"torch-distributed\",\n",
+    "    trainer=Trainer(\n",
+    "        func=train_fashion_mnist,\n",
+    "        func_args={\n",
+    "            \"epochs\": 10,\n",
+    "        },\n",
+    "        # Set how many worker Pods you want the job to be distributed into\n",
+    "        num_nodes=4,\n",
```

Comment:
I am wondering should we run this example on 2 nodes and 2 GPUs to showcase multi-proc, multi-node env ?

Reply:
That seems a good scenario for a more advanced example.
I'd be inclined to stick to single process node to keep that example as simple as possible.